docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
+Deposits must come from fully active stakes, and withdrawals return a fully 
+active stake account.
+
+Also, the deposited stake account's "credits observed" must match the destination
+account's "credits observed". Typically, this means you must wait an additional
+epoch after activation for your stake account to match up with the stake pool's account.
+
+This feature maintains fungibility of stake pool tokens. Fully activated stakes
+are not equivalent to inactive, activating, or deactivating stakes due to the
+time cost of staking. Otherwise, malicious actors can deposit stake in one state
+and withdraw it in another state without waiting.
+
+Future iterations of the stake pool program aim to improve on this limitation.
+
+## Command-line Utility
+
+The `spl-stake-pool` command-line utility can be used to experiment with SPL
+tokens.  Once you have [Rust installed](https://rustup.rs/), run:
+```sh
+$ cargo install spl-stake-pool-cli
+```
+
+Run `spl-stake-pool --help` for a full description of available commands.
+
+### Configuration
+
+The `spl-stake-pool` configuration is shared with the `solana` command-line tool.
+
+#### Current Configuration
+
+```sh
+solana config get
+Config File: ${HOME}/.config/solana/cli/config.yml
+RPC URL: https://api.mainnet-beta.solana.com
+WebSocket URL: wss://api.mainnet-beta.solana.com/ (computed)
+Keypair Path: ${HOME}/.config/solana/id.json
+```
+
+#### Cluster RPC URL
+
+See [Solana clusters](https://docs.solana.com/clusters) for cluster-specific RPC URLs
+```sh
+solana config set --url https://devnet.solana.com
+```
+
+#### Default Keypair
+
+See [Keypair conventions](https://docs.solana.com/cli/conventions#keypair-conventions)
+for information on how to setup a keypair if you don't already have one.
+
+Keypair File
+```sh
+solana config set --keypair ${HOME}/new-keypair.json
+```
+
+Hardware Wallet URL (See [URL spec](https://docs.solana.com/wallet-guide/hardware-wallets#specify-a-keypair-url))
+```sh
+solana config set --keypair usb://ledger/
+```
+
+### Example: Creating your own stake pool with 3% fee
+
+```sh
+$ spl-stake-pool create-pool --fee-numerator 3 --fee-denominator 100
+Creating mint Gmk71cM7j2RMorRsQrsyysM4HsByQx5PuDGtDdqGLWCS
+Creating pool fee collection account 3xvXPfQi2SaTkqPV9A7BQwh4GyTe2ZPasfoaCBCnTAJ5
+Creating stake pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC
+Signature: 5HdDoPssqwyLjt2QvhRbnSATZqFLGKha92zMuJiBUpKeKYKGURRV41N5ydCQxqnFjCud3xv85Z6ghErppNJzaYM8
+```
+
+The unique stake pool identifier is `3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC`.
+
+The identifier for the SPL token for staking derivatives is
+`Gmk71cM7j2RMorRsQrsyysM4HsByQx5PuDGtDdqGLWCS`. The stake pool has full control
+over the mint.
+
+The pool creator's fee account identifier is
+`3xvXPfQi2SaTkqPV9A7BQwh4GyTe2ZPasfoaCBCnTAJ5`. When users deposit SOL into the
+stake pool, the program will transfer 3% of their contribution into this account.
+
+### Example: Create validator stake account
```

Comment:
This is an "Administrator Example" right?

Reply:
This is currently a permissionless instruction since it isn't changing the stake pool in any way, but it makes more sense to have it be permissioned.  Added #1419 for that